PurchasesCoreSwift/Networking/ETagManager.swift

```diff
@@ -0,0 +1,184 @@
+//
+// Created by César de la Vega on 4/16/21.
+//
+
+import Foundation
+
+internal let ETAG_HEADER_NAME: String = "X-RevenueCat-ETag"
+
+@objc(RCETagManager) public class ETagManager: NSObject {
+    private let queue = DispatchQueue(label: "ETagManager")
+
+    private var userDefaults: UserDefaults
+
+    @objc public override init() {
+        // TODO: do they persist after reinstalling
+        let bundleID: String = Bundle.main.bundleIdentifier ?? ""
+        self.userDefaults = UserDefaults(suiteName: bundleID + ".revenuecat.etags") ?? UserDefaults.standard
+    }
+    
+    @objc public init(userDefaults: UserDefaults) {
+        self.userDefaults = userDefaults
+    }
+
+    @objc public func getETagHeader(for urlRequest: URLRequest, refreshETag: Bool = false) -> [String: String] {
+        var storedETag = ""
+        if (!refreshETag) {
+            if let storedETagAndResponse = getStoredETagAndResponse(for: urlRequest) {
+                storedETag = storedETagAndResponse.eTag
+            }
+        }
+        return [ETAG_HEADER_NAME: storedETag]
+    }
+
+    @objc public func getHTTPResultFromCacheOrBackend(with statusCode: Int,
+                                                      responseObject: [String: Any]?,
+                                                      error: Error?,
+                                                      headersInResponse: [String: Any],
+                                                      request: URLRequest,
+                                                      retried: Bool) -> HTTPResponse? {
+        let resultFromBackend = HTTPResponse(statusCode: statusCode, responseObject: responseObject)
+        if error != nil {
+            return resultFromBackend
+        }
+        
+        var eTagInResponse: String? = headersInResponse[ETAG_HEADER_NAME] as! String?
+        if ((eTagInResponse == nil)) {
+            eTagInResponse = headersInResponse[ETAG_HEADER_NAME.lowercased()] as! String?
+        }
```

Comment:
this could be simplified as
```swift
let eTagInResponse = headersInResponse[.eTagHeaderName] ?? headersInResponse[.eTagHeaderName.lowerCased()]
```